server/src/main/java/org/opensearch/gateway/remote/RemoteClusterStateCleanupManager.java

```diff
@@ -240,6 +260,11 @@ void deleteClusterMetadata(
             deleteStalePaths(clusterName, clusterUUID, new ArrayList<>(staleGlobalMetadataPaths));
             deleteStalePaths(clusterName, clusterUUID, new ArrayList<>(staleIndexMetadataPaths));
             deleteStalePaths(clusterName, clusterUUID, new ArrayList<>(staleManifestPaths));
+            try {
+                remoteRoutingTableService.deleteStaleIndexRoutingPaths(new ArrayList<>(staleIndexRoutingPaths));
+            } catch (IOException e) {
+                throw new RuntimeException(e);
```

Comment:
Will throwing exception here kill the thread running cleanup? 
I see other exceptions are caught and added to stats but not thrown. Can we check exception handling here.

Reply:
Lets add failure to stats as well